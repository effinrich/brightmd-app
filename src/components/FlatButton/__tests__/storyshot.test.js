import initStoryshots, { snapshotWithOptions } from '@storybook/addon-storyshots'

const createNodeMock = el => {
  if (el.type === 'div') {
    const mockEl = document.createElement('div')
    return mockEl
  }
  return null
}

initStoryshots({
  storyKindRegex: /^FlatButton$/,
  test: snapshotWithOptions({
    createNodeMock
  })
})
