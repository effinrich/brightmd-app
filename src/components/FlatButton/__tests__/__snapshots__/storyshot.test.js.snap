// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots FlatButton cancel 1`] = `
.c0 {
  color: rgb(101,113,121);
  display: inline-block;
  border: none;
  padding: 16px;
  background-color: transparent;
  font-size: 16px;
  height: auto;
  box-sizing: border-box;
  text-align: center;
  cursor: pointer;
  -webkit-transition: background-color 0.3s,color 0.3s;
  transition: background-color 0.3s,color 0.3s;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 0;
  background-color: rgb(197,197,197);
}

.c0 * {
  pointer-events: none;
}

.c0:hover {
  background-color: rgb(197,197,197);
  background-color: rgb(197,197,197);
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgb(216,216,216);
}

<button
  bg="transparent"
  className="c0"
  disabled={false}
  fontSize={2}
  height="auto"
  onClick={[Function]}
  p={2}
>
  This is a Cancel FlatButton
</button>
`;

exports[`Storyshots FlatButton circle button 1`] = `
.c0 {
  color: rgb(101,113,121);
  display: inline-block;
  border: none;
  padding: 16px;
  background-color: black;
  color: white;
  font-size: 16px;
  height: auto;
  box-sizing: border-box;
  text-align: center;
  cursor: pointer;
  -webkit-transition: background-color 0.3s,color 0.3s;
  transition: background-color 0.3s,color 0.3s;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 50%;
}

.c0 * {
  pointer-events: none;
}

.c0:hover {
  background-color: pink;
  color: black;
  background-color: pink;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: black;
}

<button
  bg="black"
  className="c0"
  color="white"
  disabled={false}
  fontSize={2}
  height="auto"
  onClick={[Function]}
  p={2}
>
  <svg
    fill="currentColor"
    height={34}
    size={34}
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 24 24"
    width={34}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13.5.67s.74 2.65.74 4.8c0 2.06-1.35 3.73-3.41 3.73-2.07 0-3.63-1.67-3.63-3.73l.03-.36C5.21 7.51 4 10.62 4 14c0 4.42 3.58 8 8 8s8-3.58 8-8C20 8.61 17.41 3.8 13.5.67zM11.71 19c-1.78 0-3.22-1.4-3.22-3.14 0-1.62 1.05-2.76 2.81-3.12 1.77-.36 3.6-1.21 4.62-2.58.39 1.29.59 2.65.59 4.04 0 2.65-2.15 4.8-4.8 4.8z"
    />
  </svg>
</button>
`;

exports[`Storyshots FlatButton danger 1`] = `
.c0 {
  color: rgb(101,113,121);
  display: inline-block;
  border: none;
  padding: 16px;
  background-color: transparent;
  font-size: 16px;
  height: auto;
  box-sizing: border-box;
  text-align: center;
  cursor: pointer;
  -webkit-transition: background-color 0.3s,color 0.3s;
  transition: background-color 0.3s,color 0.3s;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 0;
  background-color: rgb(202,25,0);
  color: white;
}

.c0 * {
  pointer-events: none;
}

.c0:hover {
  background-color: rgb(197,197,197);
  background-color: rgb(223,41,0);
  color: white;
  background-color: rgb(197,197,197);
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgb(223,41,0);
  color: white;
}

<button
  bg="transparent"
  className="c0"
  disabled={false}
  fontSize={2}
  height="auto"
  onClick={[Function]}
  p={2}
>
  This is a Danger FlatButton
</button>
`;

exports[`Storyshots FlatButton disabled 1`] = `
.c0 {
  color: rgb(101,113,121);
  display: inline-block;
  border: none;
  padding: 16px;
  background-color: transparent;
  font-size: 16px;
  height: auto;
  box-sizing: border-box;
  text-align: center;
  cursor: pointer;
  -webkit-transition: background-color 0.3s,color 0.3s;
  transition: background-color 0.3s,color 0.3s;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 0;
  background-color: rgb(240,240,240);
  color: rgb(204,204,204);
  cursor: not-allowed;
  pointer-events: none;
}

.c0 * {
  pointer-events: none;
}

.c0:hover {
  background-color: rgb(197,197,197);
  background-color: rgb(197,197,197);
  -webkit-text-decoration: none;
  text-decoration: none;
}

<button
  bg="transparent"
  className="c0"
  disabled={true}
  fontSize={2}
  height="auto"
  onClick={[Function]}
  p={2}
>
  This is a Disabled FlatButton
</button>
`;

exports[`Storyshots FlatButton disabled/transparent 1`] = `
.c0 {
  color: rgb(101,113,121);
  display: inline-block;
  border: none;
  padding: 16px;
  background-color: transparent;
  font-size: 16px;
  height: auto;
  box-sizing: border-box;
  text-align: center;
  cursor: pointer;
  -webkit-transition: background-color 0.3s,color 0.3s;
  transition: background-color 0.3s,color 0.3s;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 0;
  background-color: transparent;
  color: rgb(204,204,204);
  cursor: not-allowed;
  pointer-events: none;
}

.c0 * {
  pointer-events: none;
}

.c0:hover {
  background-color: rgb(197,197,197);
  background-color: rgb(197,197,197);
  -webkit-text-decoration: none;
  text-decoration: none;
}

<button
  bg="transparent"
  className="c0"
  disabled={true}
  fontSize={2}
  height="auto"
  onClick={[Function]}
  p={2}
>
  This is a Disabled transparent FlatButton
</button>
`;

exports[`Storyshots FlatButton naked story 1`] = `
.c0 {
  color: rgb(101,113,121);
  display: inline-block;
  border: none;
  padding: 16px;
  background-color: transparent;
  font-size: 16px;
  height: auto;
  box-sizing: border-box;
  text-align: center;
  cursor: pointer;
  -webkit-transition: background-color 0.3s,color 0.3s;
  transition: background-color 0.3s,color 0.3s;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 0;
}

.c0 * {
  pointer-events: none;
}

.c0:hover {
  background-color: rgb(197,197,197);
  background-color: rgb(197,197,197);
  -webkit-text-decoration: none;
  text-decoration: none;
}

<button
  bg="transparent"
  className="c0"
  disabled={false}
  fontSize={2}
  height="auto"
  onClick={[Function]}
  p={2}
>
  <div
    style={
      Object {
        "fontSize": "10px",
      }
    }
  >
    Signed in as
  </div>
  <div>
    Dani Thompson
  </div>
</button>
`;

exports[`Storyshots FlatButton primary 1`] = `
.c0 {
  color: rgb(101,113,121);
  display: inline-block;
  border: none;
  padding: 16px;
  background-color: transparent;
  font-size: 16px;
  height: auto;
  box-sizing: border-box;
  text-align: center;
  cursor: pointer;
  -webkit-transition: background-color 0.3s,color 0.3s;
  transition: background-color 0.3s,color 0.3s;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 0;
  background-color: rgb(70,55,134);
  color: white;
}

.c0 * {
  pointer-events: none;
}

.c0:hover {
  background-color: rgb(197,197,197);
  background-color: rgb(114,97,186);
  color: white;
  background-color: rgb(197,197,197);
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgb(114,97,186);
  color: white;
}

<button
  bg="transparent"
  className="c0"
  disabled={false}
  fontSize={2}
  height="auto"
  onClick={[Function]}
  p={2}
>
  This is a Primary FlatButton
</button>
`;

exports[`Storyshots FlatButton primaryInvert 1`] = `
.c0 {
  color: rgb(101,113,121);
  display: inline-block;
  border: none;
  padding: 16px;
  background-color: transparent;
  font-size: 16px;
  height: auto;
  box-sizing: border-box;
  text-align: center;
  cursor: pointer;
  -webkit-transition: background-color 0.3s,color 0.3s;
  transition: background-color 0.3s,color 0.3s;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 0;
  background-color: white;
  color: rgb(70,55,134);
  border: 2px solid rgb(70,55,134);
}

.c0 * {
  pointer-events: none;
}

.c0:hover {
  background-color: rgb(197,197,197);
  background-color: white;
  color: rgb(114,97,186);
  border: 2px solid rgb(114,97,186);
  background-color: rgb(197,197,197);
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: white;
  color: rgb(114,97,186);
  border: 2px solid rgb(114,97,186);
}

<button
  bg="transparent"
  className="c0"
  disabled={false}
  fontSize={2}
  height="auto"
  onClick={[Function]}
  p={2}
>
  This is a PrimaryInvert FlatButton
</button>
`;

exports[`Storyshots FlatButton submitting 1`] = `
.c1 {
  padding: 9px;
}

.c0 {
  color: rgb(101,113,121);
  display: inline-block;
  border: none;
  padding: 16px;
  background-color: transparent;
  font-size: 16px;
  width: 25%;
  height: auto;
  box-sizing: border-box;
  text-align: center;
  cursor: pointer;
  -webkit-transition: background-color 0.3s,color 0.3s;
  transition: background-color 0.3s,color 0.3s;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 0;
  background-color: rgb(70,55,134);
  color: white;
  background-color: rgb(240,240,240);
  color: rgb(204,204,204);
  cursor: not-allowed;
  pointer-events: none;
}

.c0 * {
  pointer-events: none;
}

.c0:hover {
  background-color: rgb(197,197,197);
  background-color: rgb(114,97,186);
  color: white;
  background-color: rgb(197,197,197);
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgb(114,97,186);
  color: white;
}

<button
  bg="transparent"
  className="c0"
  disabled={true}
  fontSize={2}
  height="auto"
  onClick={[Function]}
  p={2}
  width={0.25}
>
  <div
    className="c1"
  >
    <div
      className="loader"
    />
  </div>
</button>
`;

exports[`Storyshots FlatButton success 1`] = `
.c0 {
  color: rgb(101,113,121);
  display: inline-block;
  border: none;
  padding: 16px;
  background-color: transparent;
  font-size: 16px;
  height: auto;
  box-sizing: border-box;
  text-align: center;
  cursor: pointer;
  -webkit-transition: background-color 0.3s,color 0.3s;
  transition: background-color 0.3s,color 0.3s;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 0;
  background-color: rgb(138,226,159);
  color: white;
}

.c0 * {
  pointer-events: none;
}

.c0:hover {
  background-color: rgb(197,197,197);
  background-color: rgb(164,234,180);
  color: white;
  background-color: rgb(197,197,197);
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgb(164,234,180);
  color: white;
}

<button
  bg="transparent"
  className="c0"
  disabled={false}
  fontSize={2}
  height="auto"
  onClick={[Function]}
  p={2}
>
  This is a Success FlatButton
</button>
`;

exports[`Storyshots FlatButton text align left 1`] = `
.c0 {
  color: rgb(101,113,121);
  display: inline-block;
  border: none;
  padding: 16px;
  background-color: transparent;
  font-size: 16px;
  height: auto;
  box-sizing: border-box;
  text-align: left;
  cursor: pointer;
  -webkit-transition: background-color 0.3s,color 0.3s;
  transition: background-color 0.3s,color 0.3s;
  outline: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 0;
}

.c0 * {
  pointer-events: none;
}

.c0:hover {
  background-color: rgb(197,197,197);
  background-color: rgb(197,197,197);
  -webkit-text-decoration: none;
  text-decoration: none;
}

<button
  bg="transparent"
  className="c0"
  disabled={false}
  fontSize={2}
  height="auto"
  onClick={[Function]}
  p={2}
>
  <div
    style={
      Object {
        "fontSize": "10px",
      }
    }
  >
    Signed in as
  </div>
  <div>
    Dani Thompson
  </div>
</button>
`;
